loader_hub/jira/base.py

```diff
@@ -1,75 +1,104 @@
-from typing import List
+from typing import List, Optional, Dict, TypedDict
 
 from llama_index.readers.base import BaseReader
 from llama_index.readers.schema.base import Document
 
+
+class basic_auth(TypedDict):
+    email: str
+    api_token: str
+    server_url: str
+
+
+class oauth(TypedDict):
+    cloud_id: str
+    api_token: str
+
+
 class JiraReader(BaseReader):
     """Jira reader. Reads data from Jira issues from passed query.
 
     Args:
-        email (str): Jira email.
-        api_token (str): Jira API token.
-        server_url (str): Jira server url.
+        Optional basic_auth:{
+            "email": "email",
+            "api_token": "token",
+            "server_url": "server_url"
+        }
+        Optional oauth:{
+            "cloud_id": "cloud_id",
+            "api_token": "token"
+        }
     """
 
-    def __init__(self, email: str, api_token: str, server_url: str) -> None:
-
+    def __init__(
+        self, basic_auth: Optional[basic_auth] = None, oauth2: Optional[oauth] = None
```

Comment:
this will break backwards compatibility. Can we keep existing variable names (maybe replace type of str with Optional[str] with default kwarg = None), and then add a `basic_auth` and `oauth2` variable? if `basic_auth` or `oauth2` are specified we can do checks to make sure other variable names aren't specified

Reply:
Sure, let me add that.